ecl/hqlcpp/hqlsource.cpp

```diff
@@ -1819,19 +1819,35 @@ ABoundActivity * SourceBuilder::buildActivity(BuildCtx & ctx, IHqlExpression * e
     StringBuffer graphLabel;
     graphLabel.append(getActivityText(activityKind));
 
+    bool isFiltered = false;
+    double filterLikelihood = 1.0;
     if ((activityKind == TAKdiskread) || (activityKind == TAKcsvread) || (activityKind == TAKxmlread) || (activityKind == TAKjsonread))
     {
         graphLabel.clear();
         if (expr != tableExpr)
         {
             IHqlExpression * cur = expr;
-            bool isFiltered = false;
             bool isProjected = false;
             loop
             {
                 switch (cur->getOperator())
                 {
                 case no_filter:
+                    if (isKnownLikelihood(filterLikelihood))
+                    {
+                        unsigned childCount = cur->numChildren();
+                        for (unsigned idx=1; idx < childCount; ++idx)
+                        {
+                            double likelihood = queryLikelihood(cur->queryChild(idx));
```

Comment:
Similarly need to skip attributes.  I would be tempted to create a function queryActivityLikelihood() - especially if this code needs to be duplicated again.